### PR TITLE
feat(dialogs): replace mdl with md-dialogs

### PR DESCRIPTION
**Pull request by ksendart**
#278

### CODE REVIEW

File: src/app/dialog/dialog-service/dialog.service.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import { Observable } from 'rxjs/Rx';
+import { MdDialogRef, MdDialog, MdDialogConfig } from '@angular/material';
+import { Injectable } from '@angular/core';
+
+import { AlertDialogComponent, AlertDialogConfiguration } from './alert-dialog/alert-dialog.component';
+import { ConfirmDialogComponent, ConfirmDialogConfiguration } from './confirm-dialog/confirm-dialog.component';
+import { AskDialogConfiguration, AskDialogComponent } from './ask-dialog/ask-dialog.component';
+import { isUndefined } from 'util';
```
Comment (andrewbents) [inline src/app/dialog/dialog-service/dialog.service.ts]: let's just stick to conventional ways of checking for undefined. This one is Node module and is deprecated is the latest version

File: src/app/dialog/dialog-service/dialog.service.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+export interface BaseDialogConfiguration {
+  title?: string;
+  message: string | ParametrizedTranslation;
+  disableClose?: boolean;
+  width?: string;
+}
+
+
+@Injectable()
+export class DialogsService {
```
Comment (andrewbents) [inline src/app/dialog/dialog-service/dialog.service.ts]: let's just keep it DialogService

File: src/app/auth/logout.component.ts [hunk 3]
```diff
@@ -25,7 +23,6 @@ export class LogoutComponent implements OnInit {
       const next = this.activatedRoute.snapshot.queryParams['next'];
       const redirectionParams = next ? this.routerUtilsService.getRedirectionQueryParams(next) : {};
       this.router.navigate(['/login'], redirectionParams);
-      this.dialogService.hideAllDialogs();
       this.dialog.closeAll();
```
[the remark this answers is not in the document]
Comment (ksendart) [reply src/app/auth/logout.component.ts]: I totally replace dialogservice with new dialogservice based on md-dialog

File: src/app/app.component.scss [hunk 4]
```diff
@@ -160,3 +160,52 @@ mdl-layout-title {
...
+  justify-content: center;
+  align-items: center !important;
+  padding: 20px;
+}
+
+/deep/ .cdk-overlay-backdrop {
+  position: fixed !important;
+}
+
+/deep/ .cdk-overlay-pane {
```
[the remark this answers is not in the document]
Comment (ksendart) [reply src/app/app.component.scss]: Sure, thank you for idea. I moved all necessary styles

**Comment by wowshakhov** [comment]
There is a scrollbar in custom service offering component:

**Comment by ksendart** [comment]
fix custom service offering dialog in merge commit

**Comment by wowshakhov** [comment]
Timezone select is cut off by dialog and also focus looks weird. The minute field is focused first, but then focus disappears and tab gets focused instead.

**Comment by ksendart** [comment]
@wowshakhov  it looks strange, i think, you make create a new issue for this, because this also connects with md-tabs

**Pull request merged**